pandas/core/indexes/base.py

```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
             # coerces to object
             return self.astype(object).putmask(mask, value)
 
-    def equals(self, other) -> bool:
+    def equals(self, other: Any) -> bool:
         """
-        Determine if two Index objects contain the same elements.
+        Determine if two Index objects contain the same elements with the same order.
```

Comment:
Or we could also simplify "... contain the same elements with the same order" to "... are equal" ?

We can in a second paragraph detail on what is considered as equal